### DISCUSSION TITLE
cant use Moment type for react-hook-form values.

### DISCUSSION THREAD

**Discussion by polymoly** [react-hook-form/react-hook-form#6518] [Q&A]
i cant use Moment type (momen time package) for react hook form values. i think react hook form recursively dive into values to build literal string types. but moment trap it into a infinite loop in case of typescript. 
the error i recieved : Type of property 'caller' circularly references itself in mapped.
whats the solution?

**Comment by Joe312341** [comment]
Late reply but for anyone else coming across this, I believe it was marked as a design limitation and the preferred way is to mark dates in your form as either string or Date type and then do the conversion to moment whenever needed.

See https://github.com/react-hook-form/resolvers/issues/258
